rct_optimizations/src/rct_optimizations/pnp.cpp

```diff
@@ -1,37 +1,37 @@
-#include "rct_optimizations/experimental/pnp.h"
+#include "rct_optimizations/pnp.h"
 #include "rct_optimizations/ceres_math_utilities.h"
 #include "rct_optimizations/eigen_conversions.h"
 
 #include <ceres/ceres.h>
 
 namespace
 {
-
 struct SolvePnPCostFunc
 {
-public:
-  SolvePnPCostFunc(const rct_optimizations::CameraIntrinsics& intr, const Eigen::Vector3d& pt_in_target,
+  public:
+  SolvePnPCostFunc(const rct_optimizations::CameraIntrinsics& intr,
+                   const Eigen::Vector3d& pt_in_target,
                    const Eigen::Vector2d& pt_in_image)
     : intr_(intr), in_target_(pt_in_target), in_image_(pt_in_image)
-  {}
+  {
+  }
 
   template<typename T>
-  bool operator()(const T* const target_pose, T* const residual) const
+  bool operator()(const T *const target_q, const T *const target_t, T *const residual) const
   {
-    const T* target_angle_axis = target_pose + 0;
-    const T* target_position = target_pose + 3;
+    using Isometry3 = Eigen::Transform<T, 3, Eigen::Isometry>;
+    using Vector3 = Eigen::Matrix<T, 3, 1>;
+    using Vector2 = Eigen::Matrix<T, 2, 1>;
 
-    // Transform points into camera coordinates
-    T target_pt[3];
-    target_pt[0] = T(in_target_(0));
-    target_pt[1] = T(in_target_(1));
-    target_pt[2] = T(in_target_(2));
+    Eigen::Map<const Eigen::Quaternion<T>> q(target_q);
+    Eigen::Map<const Vector3> t(target_t);
+    Isometry3 camera_to_target = Isometry3::Identity();
+    camera_to_target = Eigen::Translation<T, 3>(t) * q;
```

Comment:
Is it possible to map `target_t` directly to an `Eigen::Translation` instead of needing to go through an `Eigen::Matrix<T,3,1>` as an intermediary?

Reply:
Good thought. I'm not sure if the pointer maps the same way, but I can test it out

Reply:
There appears to be a [bug](https://gitlab.com/libeigen/eigen/-/issues/669) with the way that Eigen maps translations from pointers into objects, so it doesn't compile. I think we'll have to leave it as a vector for now. Ultimately I would like to be able to pass the entire matrix to a local  parameterization instead of two separate objects, but I haven't worked that out yet.